frontend/src/main/scala/bloop/engine/Interpreter.scala

```diff
@@ -157,12 +161,16 @@ object Interpreter {
       state: State,
       deduplicateFailures: Boolean
   ): Task[State] = {
-    state.build.getProjectFor(cmd.project) match {
-      case Some(project) =>
-        if (!cmd.watch) runCompile(cmd, state, project, deduplicateFailures)
-        else watch(project, state)(runCompile(cmd, _, project, deduplicateFailures))
-      case None => Task.now(reportMissing(cmd.project :: Nil, state))
-    }
+    // FIXME: HERE
+    val (projects, missingProjects) = lookupProjects(cmd.project, state)
```

Comment:
I reworked this part on a commit in the branch linked. The idea here is that if there's a project that is missing, we report an error immediately and bail out from compiling. Also, I undid the change you made to the return type of `reportMissing`, it's better returning a plain state than a `Task[State]`.